### ISSUE TITLE
Invalid angle brackets in intra-doc link causes `Option::unwrap()` panic

### ISSUE THREAD

**Issue by kpreid** [rust-analyzer/rust-analyzer#10790]
Repro: Enter this code, then try to edit the file further.

```rust
/// [`<i32 as Debug>::fmt`]
pub const fn baz() {}
```

This is not (currently) valid intra-doc link syntax, but it probably shouldn't cause rust-analyzer to panic every time a character is typed.

```
Panic context:
> 
version: 73668334f 2021-11-15 stable
request: textDocument/semanticTokens/full/delta SemanticTokensDeltaParams {
    work_done_progress_params: WorkDoneProgressParams {
        work_done_token: None,
    },
    partial_result_params: PartialResultParams {
        partial_result_token: None,
    },
    text_document: TextDocumentIdentifier {
        uri: Url {
            scheme: "file",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: None,
            port: None,
            path: "/Users/kpreid/Projects/rust/scratchpad/src/bin/ra-crash.rs",
            query: None,
            fragment: None,
        },
    },
    previous_result_id: "31",
}

thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', crates/hir/src/attrs.rs:150:84
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: hir::attrs::resolve_doc_path
   4: ide::doc_links::resolve_doc_path_for_def
   5: ide::syntax_highlighting::inject::doc_comment
   6: ide::syntax_highlighting::traverse
   7: ide::syntax_highlighting::highlight
   8: std::panicking::try
   9: rust_analyzer::handlers::handle_semantic_tokens_full_delta
  10: std::panicking::try
  11: <F as threadpool::FnBox>::call_box
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[Error - 11:44:37 AM] Request textDocument/semanticTokens/full/delta failed.
  Message: server panicked: called `Option::unwrap()` on a `None` value
  Code: -32603 
```

Editor: VS Code

**Comment by Veykril** [comment]
Ye this unwrap here is just wrong:
https://github.com/rust-analyzer/rust-analyzer/blob/32f425d801ad46c888a3a3639c5e2edbff4f3811/crates/hir/src/attrs.rs#L150

**Comment by lnicola** [comment]
Fixed in #10795
